sheets-core/src/commonMain/kotlin/com/dokar/sheets/layout/CoreBottomSheetLayout.kt

```diff
@@ -0,0 +1,254 @@
+        SheetNestedScrollConnection(state, coroutineScope)
+    }
+
+    val dimAlpha = min(maxDimAmount, max(0f, state.dimAmount))
+
+    val density = LocalDensity.current
+
+    val initialOffsetY = with(density) { 180.dp.toPx() }
+
+    val contentAlpha = remember(state) { Animatable(0f) }
+
+    var gestureDownPos by remember { mutableStateOf(Offset.Zero) }
+
+    val sheetLayoutState = rememberSheetContentLayoutState()
+
+    val topInset = WindowInsets.Companion.statusBars
+
+    var contentRect by remember { mutableStateOf(Rect.Zero) }
+
+    SideEffect {
+        state.peekHeight = peekHeight
+        state.forceSkipPeeked = skipPeeked
+        state.maxDimAmount = maxDimAmount
+    }
+
+    LaunchedEffect(state) {
+        launch {
+            snapshotFlow { state.value }
+                .distinctUntilChanged()
+                .filter { it == BottomSheetValue.Collapsed }
+                .collect { contentAlpha.snapTo(0f) }
+        }
+    }
+
+    DisposableEffect(state) {
+        onDispose {
+            state.visible = false
+            state.contentHeight = 0
+            coroutineScope.launch {
+                state.stopAnimations()
+            }
+        }
+    }
+
+    Box(
+        modifier = Modifier
+            .then(
+                if (behaviors.isModal) {
+                    Modifier
+                        .fillMaxSize()
+                        .drawBehind { drawRect(color = dimColor.copy(alpha = dimAlpha)) }
+                        .nestedScroll(nestedScrollConnection)
+                        .clickable(
+                            interactionSource = remember { MutableInteractionSource() },
+                            indication = null,
+                            onClick = {
+                                if (!behaviors.collapseOnClickOutside) {
+                                    return@clickable
+                                }
+                                if (state.dragVelocity < 1000f
+                                    && sheetLayoutState.isWithinContentBounds(gestureDownPos)
+                                ) {
+                                    // Swiping up but the tap event was triggered, ignore
+                                    return@clickable
+                                }
+                                coroutineScope.launch { state.collapse() }
+                            },
+                        )
+                        .detectPointerPositionChanges(
+                            key = state,
+                            onDown = {
+                                state.resetVelocity()
+                                gestureDownPos =
+                                    it.copy(y = it.y - topInset.getTop(density))
+                            },
+                            onPositionChanged = { state.addVelocity(it.y) },
+                        )
+                } else {
+                    Modifier
```

Comment:
Is it safe to ignore all these Modifiers when isModal is false? Especially, the `nestedScroll()`?

Reply:
Works perfectly fine in my cases, maybe not tested any corner cases, but not shure.

Reply:
Have you tried using it with a lazy list in the sheet content? If it still works we are probably fine without `nestedScroll`.

Reply:
https://github.com/user-attachments/assets/a2a39448-9699-4e5c-983a-2620cecfaf9f

Yea, it still works.